public/app/features/explore/state/utils.ts

```diff
@@ -107,12 +109,16 @@ export async function loadAndInitDatasource(
 }
 
 export function getUrlStateFromPaneState(pane: ExploreItemState): ExploreUrlState {
+  const prunedPanelsState = omitBy(pane.panelsState, isEmpty);
+
   return {
     // datasourceInstance should not be undefined anymore here but in case there is some path for it to be undefined
     // lets just fallback instead of crashing.
     datasource: pane.datasourceInstance?.name || '',
     queries: pane.queries.map(clearQueryKeys),
     range: toRawTimeRange(pane.range),
+    // don't include panelsState in the url unless a piece of state is actually set
+    panelsState: isEmpty(prunedPanelsState) ? undefined : prunedPanelsState,
```

Comment:
panelsState could get unwieldy and take up a lot of bytes in the url, so the idea is to only encode panel state in the url where at least one key is not undefined.